tests/test_connector.rs

```diff
@@ -0,0 +1,351 @@
+        "ft_balance_of",
+        json!({ "account_id": acc }).to_string().as_bytes(),
+    );
+    String::from_utf8(balance.unwrap())
+        .unwrap()
+        .parse()
+        .unwrap()
+}
+
+fn get_eth_balance(master_account: &UserAccount, address: EthAddress) -> u128 {
+    let address = hex::encode(address);
+    let balance = master_account.view(
+        CONTRACT_ACC.to_string(),
+        "ft_balance_of_eth",
+        json!({ "address": address }).to_string().as_bytes(),
+    );
+    String::from_utf8(balance.unwrap())
+        .unwrap()
+        .parse()
+        .unwrap()
+}
+
+fn total_supply(master_account: &UserAccount) -> u128 {
+    let balance = master_account.view(CONTRACT_ACC.to_string(), "ft_total_supply", &[]);
+    String::from_utf8(balance.unwrap())
+        .unwrap()
+        .parse()
+        .unwrap()
+}
+
+fn total_supply_near(master_account: &UserAccount) -> u128 {
+    let balance = master_account.view(CONTRACT_ACC.to_string(), "ft_total_supply_near", &[]);
+    String::from_utf8(balance.unwrap())
+        .unwrap()
+        .parse()
+        .unwrap()
+}
+
+fn total_supply_eth(master_account: &UserAccount) -> u128 {
+    let balance = master_account.view(CONTRACT_ACC.to_string(), "ft_total_supply_eth", &[]);
+    String::from_utf8(balance.unwrap())
+        .unwrap()
+        .parse()
+        .unwrap()
+}
+
+#[test]
+fn test_near_deposit_balance_total_supply() {
+    let (master_account, _contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_near(&master_account);
+
+    let balance = get_near_balance(&master_account, DEPOSITED_RECIPIENT);
+    assert_eq!(balance, DEPOSITED_AMOUNT - DEPOSITED_FEE);
+
+    let balance = get_near_balance(&master_account, CONTRACT_ACC);
+    assert_eq!(balance, DEPOSITED_FEE);
+
+    let balance = total_supply(&master_account);
+    assert_eq!(balance, DEPOSITED_AMOUNT);
+
+    let balance = total_supply_near(&master_account);
+    assert_eq!(balance, DEPOSITED_AMOUNT);
+
+    let balance = total_supply_eth(&master_account);
+    assert_eq!(balance, 0);
+}
+
+#[test]
+fn test_deposit_eth_and_near() {
+    let (master_account, _contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_near(&master_account);
+    let (master_account, _contract) = init(EVM_CUSTODIAN_ADDRESS);
+    call_deposit_eth(&master_account);
+}
+
+#[test]
+fn test_eth_deposit_balance_total_supply() {
+    let (master_account, _contract) = init(EVM_CUSTODIAN_ADDRESS);
+    call_deposit_eth(&master_account);
+
+    let balance = get_eth_balance(&master_account, validate_eth_address(RECIPIENT_ETH_ADDRESS));
+    assert_eq!(balance, DEPOSITED_EVM_AMOUNT - DEPOSITED_EVM_FEE);
+
+    let balance = total_supply(&master_account);
+    assert_eq!(balance, DEPOSITED_EVM_AMOUNT);
+
+    let balance = total_supply_eth(&master_account);
+    assert_eq!(balance, DEPOSITED_EVM_AMOUNT);
+
+    let balance = total_supply_near(&master_account);
+    assert_eq!(balance, 0);
+}
+
+#[test]
+fn test_withdraw_near() {
```

Comment:
As part of this test I think we should check the balance of `RECIPIENT_ETH_ADDRESS` too.

Reply:
@birchmd as Alex Shevchenko announced that logic will be changed.